### PR TITLE
Add generic SSE handler

### PR DESCRIPTION
**Pull request by colinking**
## Description

Adds a new generic handler for SSE endpoints. Switches logs to use it.

## Test plan

Ran a task locally and confirmed that logs render for (1) active runs and (2) historical runs.

### CODE REVIEW

**Comment by colinking** [comment]
Current dependencies on/for this PR:
* main
  * **PR #579** <a href="https://app.graphite.dev/github/pr/airplanedev/cli/579" target="_blank">Graphite</a> 
    * **PR #592** <a href="https://app.graphite.dev/github/pr/airplanedev/cli/592" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/airplanedev/cli/592?utm_source=stack-comment).

File: pkg/server/handlers.go [hunk 1]
```diff
@@ -57,6 +58,42 @@ func Handler[Resp any](state *State,
...
 
+func HandlerSSE[Resp any](state *State, f func(ctx context.Context, state *State, r *http.Request, flush func(resp Resp) error) error) http.HandlerFunc {
+	return Wrap(func(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+		w.Header().Set("Content-Type", "text/event-stream")
+		w.Header().Set("Cache-Control", "no-cache")
+		w.Header().Set("Connection", "keep-alive")
+
+		flusher, _ := w.(http.Flusher)
+		flush := func(resp Resp) error {
+			if _, err := fmt.Fprintf(w, "data: "); err != nil {
```
Comment (calufornia) [inline pkg/server/handlers.go:69]:
```suggestion
			if _, err := fmt.Fprint(w, "data: "); err != nil {
```

File: pkg/server/handlers.go [hunk 2]
```diff
@@ -57,6 +58,42 @@ func Handler[Resp any](state *State,
...
+				return errors.Wrap(err, "serializing event header")
+			}
+			e := json.NewEncoder(w)
+			e.SetEscapeHTML(false)
+			if err := e.Encode(resp); err != nil {
+				return errors.Wrap(err, "serializing event data")
+			}
+			if _, err := fmt.Fprintf(w, "\n"); err != nil {
```
Comment (calufornia) [inline pkg/server/handlers.go:77]:
```suggestion
			if _, err := fmt.Fprint(w, "\n"); err != nil {
```

File: pkg/server/handlers.go [hunk 3]
```diff
@@ -57,6 +58,42 @@ func Handler[Resp any](state *State,
...
+func HandlerSSE[Resp any](state *State, f func(ctx context.Context, state *State, r *http.Request, flush func(resp Resp) error) error) http.HandlerFunc {
+	return Wrap(func(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+		w.Header().Set("Content-Type", "text/event-stream")
+		w.Header().Set("Cache-Control", "no-cache")
+		w.Header().Set("Connection", "keep-alive")
+
+		flusher, _ := w.(http.Flusher)
+		flush := func(resp Resp) error {
+			if _, err := fmt.Fprintf(w, "data: "); err != nil {
+				return errors.Wrap(err, "serializing event header")
```
Comment (calufornia) [inline pkg/server/handlers.go:70]:
```suggestion
				return errors.Wrap(err, "serializing event field")
```

**Review by calufornia — APPROVED** [review]
thanks for this!

**Pull request merged**